ginrummy/GinRummyEAAIorigin/GinRummyEAAI/TurnStatesDataCollector.java

```diff
@@ -39,7 +44,9 @@
 	
 	/**
 	 * The multi-dimentional array to store the game data in arrays of short[][], where
-	 * playData[0] is the model's input
+	 * playData[0] is the faceupCard's input
+	 * playData[1] is the drawCard's input, if it appears that opponent pick up faceupCard
+	 * playData[2] is the discardCard's input
```

Comment:
## data length = 5:

* Vector of cards the opponent discarded this turn
* Vector of cards that the opponent does not care about (not picking up from discard pile) this turn.
* Vector of cards that the opponent does care and pick up from the discard pile.
* Cards that are on this player's hand and in the discard pile.
* Output (Opponent hand)